airsonic-main/src/main/java/org/airsonic/player/service/SecurityService.java

```diff
@@ -166,13 +166,24 @@ public User getUserByEmail(String email) {
      * Returns whether the given user has administrative rights.
      */
     public boolean isAdmin(String username) {
-        if (User.USERNAME_ADMIN.equals(username)) {
-            return true;
-        }
         User user = getUserByName(username);
         return user != null && user.isAdminRole();
     }
 
+    /**
+     * Returns username with admin rights.
+     *
+     * @return username of admin user
+     */
+    public String getAdminUsername() {
+        for (User user : userDao.getAllUsers()) {
+            if (user.isAdminRole()) {
+                return user.getUsername();
+            }
+        }
+        return null;
```

Comment:
I'd throw an exception here instead of returning null.

Reply:
I don't really like returning exceptions from simple lookup functions, especially when returning null for "not found" is such a universally accepted (and expected) pattern.  I think it restricts potential use cases unnecessarily and violates separation of concerns.  Why should this function need to enforce that the higher level system thinks not having a user with a particular role is an exception-worthy problem?

In any case, I believe that when this function returns null, the calling site will still generate an exception when it tries to use the null value, and does so without this function needing to have any concept of higher level user permissions concerns.

I also didn't want to rule out the possibility that someone might want to completely lock down their installation by disabling all admin accounts (even though doing so would require directly changing the db).

Returning null also allows for if/when someone changes the code to implement your suggestion number 3 above (making the owner of imported playlists nullable), although I admit that at that point this function might just be removed entirely, but who knows?

Reply:
@muff1nman If throwing an exception is a requirement to get your approval, say so and I will change.

Reply:
> I don't really like returning exceptions from simple lookup functions, especially when returning null for "not found" is such a universally accepted (and expected) pattern.

In Java land its not universally accepted. In other languages, go, c/c++, python, yes I would agree.

>  Why should this function need to enforce that the higher level system thinks not having a user with a particular role is an exception-worthy problem?

Its in the SecurityService and warrants a bit more checking around assumptions to ensure we fail fast. With this change we still have an assumption that there should always be at least one admin user. When that assumption changes, we should get this exception and we can change the code to deal with the new assumptions then.

> I also didn't want to rule out the possibility that someone might want to completely lock down their installation by disabling all admin accounts (even though doing so would require directly changing the db)

In the case the user does this anything that uses this function should rightly be broken. If the calling code can be robust in the scenario of no admin users, its on the function to handle the exception. 

> Returning null also allows for if/when someone changes the code to implement your suggestion number 3 above (making the owner of imported playlists nullable), although I admit that at that point this function might just be removed entirely, but who knows?

And when that happens we can change/remove this code too :)

Reply:
It should be noted that the other user lookup functions already in SecurityService (e.g. `getUserByName()`, `getUserByEmail()`) also use returning null to indicate "not found", so throwing an exception rather than returning null would be inconsistent with those.